src/IotTelemetrySimulator/TelemetryValues.cs

```diff
@@ -73,12 +84,12 @@ public TelemetryValues(IList<TelemetryVariable> variables)
                     if (previous != null && previous.TryGetValue(val.Name, out var prevValue))
                     {
                         var step = val.Step ?? 1;
-                        var maxThres = val.Max ?? int.MaxValue;
+                        var maxThres = val.Max ?? long.MaxValue;
 
                         switch (prevValue)
                         {
                             case int value when value + step > maxThres:
```

Comment:
Is this ever going to be true [(example here)](https://dotnetfiddle.net/dXnsKu)?

We add two integers, generating another int. The int can't be greater than long.MaxValue.
We should use the same type everywhere. If we want to change to long we should change the actual value type as well.